pkg/handler/handler.go

```diff
@@ -34,6 +33,16 @@ func (h *Handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+func (h *Handler) parse(r *http.Request) (hooks.PushEvent, error) {
+	h.log.Info("processing hook request")
+	data, err := ioutil.ReadAll(r.Body)
```

Comment:
Not for this branch, but I had a `// TODO: LimitReader` on the original ReadAll calls.

This is a reference to https://golang.org/pkg/io/#LimitReader which I think should be applied to guard against bad actors.